engine/src/io/data_provider/DummyProvider.h

```diff
@@ -34,27 +34,27 @@ class dummy_data_provider : public data_provider {
 		// does nothing
 	}
 
-	data_handle get_next() {
+	data_handle get_next(bool open_file = true) {
 		data_handle handle;
 		handle.fileHandle = nullptr;
 		return handle;
 	}
 
-	data_handle get_first() {
-		data_handle handle;
-		handle.fileHandle = nullptr;
-		return handle;
-	}
-
-
 	std::vector<std::string> get_errors() { return {}; }
 
-	std::string get_current_user_readable_file_handle() { return ""; }
-
-	std::vector<data_handle> get_all() { return {}; }
-
+	/**
+	 * Tries to get up to num_files data_handles. We use this instead of a get_all() because if there are too many files, 
+	 * trying to get too many file handles will cause a crash. Using get_some() forces breaking up the process of getting file_handles.
+	 */
+	std::vector<data_handle> get_some(std::size_t num_files, bool open_file = true) { return {}; }
```

Comment:
When you made this change you need to ask yourself HOW am I deciding how many files at at time. What was caussing the crashes  you were trying to resolve? Were you running out of file handles? Maybe you just need to reuse them? This seems to be a way to stop the crashes right now but it doesn't tell us if we solved the underlying issue. Also this needs to be moved into its own PR it is a bug and not related to this functionality.

Reply:
You are correct, if should be in its own PR and it is https://github.com/BlazingDB/blazingsql/pull/532
This PR has both because I wanted the user that requested this to test out both things.
That same user will tell us if this will indeed fix the issue we were seeing.
Either way, we do know that allowing the engine to open an unlimited number of file handles at a time will cause problems, so I think that we do want to implement a way to limit the number of file handles that are opened at a time and this is one way to deal with that.
Now as for deciding how many to open at a time, I dont not have a good answer to that. I am open to suggestions